File: 백준/Silver/1181. 단어 정렬/단어 정렬.cc
#include <iostream>
#include <algorithm>

using namespace std;
string a[20000];
int n;
bool compare(string a, string b) {
    if (a.length() == b.length()){
        return a < b;
    } else {
        return a.length() < b.length();
    }
}

int main() {
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> a[i];
    }
    sort(a, a + n, compare);
    for(int i = 0; i<n; i++) {
        if(a[i] == a[i-1]){
            continue;
        }
        cout << a[i]<< endl;
    };
    return 0;
}

File: 백준/Silver/1181. 단어 정렬/단어 정렬.java
import java.util.*;
import java.io.*;

public class Main {
	
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		String list[]=new String[N];
		for(int i=0; i<N; i++) 
			list[i] = br.readLine();
		
		Arrays.sort(list, new Comparator<String>(){
			@Override
			public int compare(String o1, String o2) {
				if(o1.length()==o2.length())
					return o1.compareTo(o2);
				else
					return o1.length()-o2.length();
			}
		});
		for(int i=0; i<N; i++) {
			if(i!=0) {
				if(list[i].compareTo(list[i-1])!=0)
					System.out.println(list[i]);
			}
			else
				System.out.println(list[i]);
		}
	}
}